update_database_scripts/unpackDatabaseCommandLine.R

```diff
@@ -591,6 +591,27 @@ if (is_ebi_reachable()) {
         mutate(unique_beta_units = n_distinct(tolower(beta_unit))) %>% # create column counting number of unique beta units per group
         filter(unique_beta_units==1) %>% # remove all groups with more than 1 unique beta unit
         dplyr::select(-unique_beta_units) # remove the unique beta units column
+      
+      # resolve snps that have differences in rounding which leads to duplicates
+      studyData <- studyData %>%
+        mutate(rounded_beta = floor(betaValue) + signif(betaValue - floor(betaValue), 2)) %>% # create a rounded betaValue column for comparing (2 sig figs after the decimal)
+        mutate(rounded_odds = floor(or_per_copy_number) + signif(or_per_copy_number - floor(or_per_copy_number), 2)) %>% # create a rounded oddsRatio column for comparing (2 sig figs after the decimal)
+        group_by(variant_id, risk_allele, pvalue, trait, pvalue_description, beta_description, beta_unit, ogValueTypes, studyID, rounded_beta, rounded_odds) %>%
+        mutate(num_repeated_snps = dplyr::n()) %>%
+        ungroup() %>%
+        mutate(or_per_copy_number = ifelse(num_repeated_snps > 1, rounded_odds, or_per_copy_number)) %>%
+        mutate(betaValue = ifelse(num_repeated_snps > 1, rounded_beta, betaValue)) %>%
+        distinct(variant_id, risk_allele, pvalue, trait, pvalue_description, beta_description, beta_unit, ogValueTypes, studyID, rounded_beta, rounded_odds, .keep_all = TRUE) %>%
+        dplyr::select(-rounded_beta, -rounded_odds, -num_repeated_snps) # remove the rounded columns and num_repeated_snps column
+      
+      # remove the duplicate snps that we can't resolve
+      studyData <- studyData %>%
+        group_by(variant_id, risk_allele, trait, pvalue_description, beta_description, ogValueTypes, studyID) %>%
+        mutate(num_repeated_snps = dplyr::n()) %>%
+        ungroup() %>%
+        group_by(trait, pvalue_description, beta_description, ogValueTypes, studyID) %>%
+        filter(num_repeated_snps==1) %>%
```

Comment:
Does this get rid of ALL the snps from a study (by the unique key of trait, pValueAnnotation, betaAnnotation, ogValueTypes, studyID) or just the duplicated ones? We need to make sure that ALL the snps from that combination are removed.

Reply:
Fixed. The whole study is now removed.